### PR TITLE
[ENG-6118] fix total count

### PR DESCRIPTION
**Pull request by gnordhielm**
Fixes an issue where the total count returned by paginated resource did not match the most current data we'd fetched from the server.

The fix itself is just a couple lines in `UNSTABLE__usePaginatedResource.ts`, but I wanted to restore unit tests, which accounts for the bulk of this PR.

@lukwallace, can you provide me repro instructions so I can make sure this is fixed in the UI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,19 +27,24 @@
     "checkPackage": "yarn package-check",
     "format": "prettier --write \"./**/*.{js,jsx,json,ts,tsx}\" && sort-package-json \"package.json\" \"./*/package.json\"",
     "start": "yarn test --watch",
-    "test": "echo 'tests disabled'"
+    "test": "NODE_OPTIONS=--experimental-vm-modules NODE_ENV=\"test\" jest --no-watchman"
```
Comment (mike591) [inline package.json:30]: Getting some import issues when trying to run the tests:

image

**Review by mike591 — COMMENTED** [review]
The count fix seems straightforward but some tests are failing for me.

I vaguely remember seeing the count issue when debugging with Wallace but forgot how we ended up resolving it. Will leave it to @lukwallace to approve PR/provide repro instructions.

File: package.json (hunk 1 above)
Comment (gnordhielm) [reply package.json:30]: Are you 100% sure you npm installed before running?

Comment (mike591) [reply package.json:30]: oh, I ran `yarn`. I could try `npm install`.

Comment (mike591) [reply package.json:30]: So I tried it again with `yarn` before trying `npm install` and saw that it did something. `yarn test` now works for me. Maybe I really did forget to run `yarn` 🤷 .

**Review by lukwallace — APPROVED** [review]
The code change itself looks innocuous to me. As for testing details -- how I originally found the error was that I had a `console.log` for the values of the `.use` call and noticed some strangeness:

* Add console.log
```
const [list, count] = paginatedResource.use()
console.log('heres my log', list, count);
```
* Load the page with the table in it and have it populate its rows
* In another window/tab emulate user activity by uploading data that would produce new rows in the table (in my case I was doing files w/ the  lender uploads page).
* Return to the page with the table in the UI and scroll down.
* Look in the network tab to see that calls for both the count and paginated queries are being made
* Scroll down far enough to see the new data that you introduced
* Look in the console logs to see that the `list` variable has been updated with new items, but the `count` variable remains the same as the on-load value.

Also, sorry for the late reply! I'm going to be OOO until Friday so I've been trying to close the loop on things before leaving (turns out I had a lot of loops). Haven't actually had the time to test it out yet, but happy to take an in-depth look when I get back.

**Comment by gnordhielm** [comment]
Going to merge b/c of approval. Will conduct some tests in admin after deploy.

**Pull request merged**